Sources/SwiftGen/Path+AppSupport.swift

```diff
@@ -36,4 +36,15 @@ let templatesRelativePath: String = {
 private final class BundleToken {}
 
 let appSupportTemplatesPath = Path.applicationSupport + "SwiftGen/templates"
-let bundledTemplatesPath = Path(ProcessInfo.processInfo.arguments[0]).parent() + templatesRelativePath
+let binaryPath: Path = {
+  var binaryPath = Path(ProcessInfo.processInfo.arguments[0])
+  do {
+    while binaryPath.isSymlink {
+      binaryPath = try binaryPath.symlinkDestination()
+    }
```

Comment:
Playing the devil's advocate here, isn't there a risk of a symlink reference cycle on the filesystem, that would make this kind of loop run forever? In such case, does this code automatically throw already (the `NSFileManager` that is used under the hood by `PathKit` ends up throwing its own `Error`)? Or is it our responsibility to add a max redirection threshold?

```
$ ln -s a b
$ ls -s b a
$ cat a
cat: a: Too many levels of symbolic links
```

Reply:
How could your binary be a symlink to itself 🤔 
If this code is called, this means you're executing `swiftgen` binary, so `arguments[0]` can only reference the `swiftgen` binary in the end, even if you have multiple redirections 😉

Reply:
😅 very good point 😂

Also I'm wondering, isn't there any other means to get the real path to the current process rather than rely on `args[0]`? Maybe there's some property on `ProcessInfo` or whatnot? 🤔

Reply:
I did not find any on [`ProcessInfo`](https://developer.apple.com/documentation/foundation/processinfo). I tried [`NSRunningApplication.executableURL`](https://developer.apple.com/documentation/appkit/nsrunningapplication/1531062-executableurl) but this does not seem to work. It gives me a `nil` result.

Reply:
Ok, let's keep that symlink resolution loop then